From 04ac29bc736e92fc78aa8fb80d17c4824b10c086 Mon Sep 17 00:00:00 2001
From: Alexei Colin <acolin@isi.edu>
Date: Sat, 20 Mar 2021 01:32:06 +0000
Subject: [PATCH 2/2] config: petsc4py: honor DESTDIR make var

Top-level makefile honors destination directory (which is a path where
files are to be staged, orthogonal to the prefix path). The destination
path is specified via DESTDIR variable: `make DESTDIR=foo install`.

But, before this commit, the makefile target `petsc4py-install` did not
honor DESTDIR and ends up staging files into /, which created
counter-intuitive behavior that's hard to work around in packaging
scripts.

If DESTDIR is not set, then installation command is unchanged, exactly
as it was before this commit.

Signed-off-by: Alexei Colin <acolin@isi.edu>
---
 config/BuildSystem/config/packages/petsc4py.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/config/BuildSystem/config/packages/petsc4py.py b/config/BuildSystem/config/packages/petsc4py.py
index a954ffda34..b5700023f2 100644
--- a/config/BuildSystem/config/packages/petsc4py.py
+++ b/config/BuildSystem/config/packages/petsc4py.py
@@ -73,7 +73,8 @@ class Configure(config.package.Package):
     self.addMakeRule('petsc4pyinstall','', \
                        ['@echo "*** Installing petsc4py ***"',\
                           '@(MPICC=${PCC} && export MPICC && cd '+self.packageDir+' && \\\n\
-           '+newdir+archflags+self.python.pyexe+' setup.py install --install-lib='+installLibPath+')  || \\\n\
+           '+newdir+archflags+self.python.pyexe+' setup.py install --install-lib='+installLibPath+' \\\n\
+               $(if $(DESTDIR),--root=\'$(DESTDIR)\') ) || \\\n\
              (echo "**************************ERROR*************************************" && \\\n\
              echo "Error building petsc4py." && \\\n\
              echo "********************************************************************" && \\\n\
-- 
2.30.0

